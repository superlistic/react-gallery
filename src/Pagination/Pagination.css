.pagination {
  position: fixed;
  display: flex;
  justify-content: space-between;
  align-content: center;
  bottom: 0rem;
  width: 100%;
  height: calc(33% + 5rem);
  pointer-events: none;
}

.pagination__arrow {
  font-size: calc(1rem + 2vmin);
  cursor: pointer;
  margin: 0 1rem;
  pointer-events: all;
  color: var(--clr-dark);
  border-radius: 5rem;
  background-color: var(--clr-trans);
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.575);
}
