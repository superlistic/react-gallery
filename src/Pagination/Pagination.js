import React from 'react';
import '../main.css';
import './Pagination.css';
import ArrowLeftIcon from '@material-ui/icons/ArrowLeft';
import ArrowRightIcon from '@material-ui/icons/ArrowRight';

const Pagination = ({ page, setPage, total }) => {
  const handlePagination = (num) => {
    const newPage = page + num;
    if (total > newPage && newPage > 0) {
      setPage(newPage);
      document.documentElement.scrollTop = 0;
    } else {
      console.log('OUT OF BOUNDS');
    }
  };
  return (
    <div className="pagination">
      <ArrowLeftIcon
        style={{ fontSize: '5.5rem' }}
        className="pagination__arrow"
        onClick={() => handlePagination(-1)}
      />
      <ArrowRightIcon
        style={{ fontSize: '5.5rem' }}
        className="pagination__arrow"
        onClick={() => handlePagination(1)}
      />
    </div>
  );
};

export default Pagination;
