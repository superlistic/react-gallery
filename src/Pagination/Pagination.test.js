import React from 'react';
import Pagination from './Pagination';
import renderer from 'react-test-renderer';

test('renders elements', () => {
  const tree = renderer.create(<Pagination />).toJSON();
  expect(tree).toMatchSnapshot();
});
