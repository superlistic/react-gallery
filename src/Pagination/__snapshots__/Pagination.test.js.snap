// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders elements 1`] = `
<div
  className="pagination"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root pagination__arrow"
    focusable="false"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "5.5rem",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M14 7l-5 5 5 5V7z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root pagination__arrow"
    focusable="false"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "5.5rem",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M10 17l5-5-5-5v10z"
    />
  </svg>
</div>
`;
